gammapy/time/lombscargle_gammapy.py

```diff
@@ -0,0 +1,171 @@
+    lg = np.log(pdf)
+    mask = np.isfinite(lg)
+    nll = -lg[mask].sum()
+    return nll
+
+def _cvm(param, *args):
+    """Cramer-von-Mises distance minimization
+    """
+    a, b = param
+    data = args[0]
+    ordered_data = np.sort(data, axis=None)
+    for n in range(len(data)):
+        sumbeta =+ (beta.cdf(ordered_data[n], a, b, loc=0, scale=1) - (n - 0.5) / len(data))**2.0
+    cvm_dist = (1. / len(data)) * sumbeta + 1. / (12 * (len(data)**2.))
+    mask = np.isfinite(cvm_dist)
+    cvm = cvm_dist[mask]
+    return cvm
+
+def _window_function(t):
+    least_exponent = str(t[0])[::-1].find('.')
+    dt = 10**-least_exponent
+    t_length = (np.max(t) - np.min(t)) / dt
+    t_win = np.linspace(dt, np.around(np.max(t), decimals=abs(least_exponent+1)), t_length, endpoint=True)
+    window = np.zeros(len(t_win))
+    window[np.searchsorted(t_win, np.around(t, decimals=abs(least_exponent+1)))] = 1
+
+    return t_win, window
+
+def _bootstrap(t, mag, dmag, freq, N_bootstraps):
+    rand = np.random.RandomState(42)
+    FAP = 0.05
+    N_bootstraps = 100
+    max_periods = np.empty(N_bootstraps)
+    for m in range(N_bootstraps):
+        ind = rand.randint(0, len(mag), len(mag))
+        PLS_boot = LombScargle(t, mag[ind], dmag[ind]).power(freq)
+        max_periods[m] = np.max(PLS_boot)
+
+    return max_periods
+
+def _grid(t, K):
+    max_period = (np.max(t) - np.min(t)) / 10 # see Halpern
+    min_period = 2 * np.median(t[1:-1] - t[0:-2])    
+    d_p = min_period / K    
+    n_periods = max_period / d_p
+    periods = np.linspace(min_period, max_period, n_periods)
+    grid = 1. / periods
+
+    return grid
+
+def _significance(t, mag, dmag, freq, PLS, FAP, N_bootstraps):
+    percentile = 1 - FAP
+    # compute pre-defined Beta-Distribution
+    a = (3-1)/2
+    b = (len(t)-3)/2
+    quant_pre = beta.ppf(percentile**(1./len(freq)), a, b)
+    
+    # compute CvM-minimization-fitted Beta-Distribution
+    # initial guess for CvM-minimization
+    theta_1 = -1. * (np.mean(PLS) * (-np.mean(PLS) + np.mean(PLS)**2 + np.var(PLS))) / (np.var(PLS))
+    if theta_1 < 0:
+        theta_1 = 0.00001
+    theta_2 = (theta_1 - theta_1 * np.mean(PLS)) / np.mean(PLS)
+    if theta_2 < 0:
+        theta_2 = 0.00001
+    x0 = [theta_1, theta_2]
+    cvm_minimize = optimize.fmin(_cvm, [theta_1, theta_2], args=(PLS,))
+    quant_cvm = beta.ppf(percentile**(1./len(freq)), cvm_minimize[0], cvm_minimize[1])
+    
+    # compute NLL-minimization-fitted Beta-Distribution
+    nll_minimize = optimize.fmin(_nll, [a, b], args=(PLS,))
+    quant_nll = beta.ppf(percentile**(1./len(freq)), nll_minimize[0], nll_minimize[1])
+
+    # bootstrap
+    max_periods = _bootstrap(t, mag, dmag, freq, N_bootstraps)
+    sig = np.percentile(max_periods, 100*percentile, axis=0)
+
+    return quant_pre, quant_cvm, quant_nll, sig
+
+def lomb_scargle(t, mag, dmag, K, N_bootstraps, FAP):
+    """This function processes the lomb-scargle-algorithmus in autopower mode
```

Comment:
So this is the most important function for users.
Please write the docstring here with `Parameters` and `Returns` section documenting the inputs / outputs.
And maybe, if you have additional info to share (e.g. a link to a paper with the method), add that in the docstring as well.

Here's an example for the Numpy docstring format  we use:
* Docstring: http://docs.gammapy.org/en/latest/_modules/gammapy/stats/poisson.html#significance
* Rendered HTML version: http://docs.gammapy.org/en/latest/api/gammapy.stats.significance.html